File: backend/app/schemas/issue.py
from pydantic import BaseModel
from datetime import date, datetime


class IssueCreate(BaseModel):
    title: str
    description: str
    type: str  # action, issue, decision
    priority: str  # Alta, Media, Baja
    report_date: date
    commitment_date: date | None = None
    responsible_id: int | None = None


class IssueUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    type: str | None = None
    priority: str | None = None
    status: str | None = None
    resolution: str | None = None
    commitment_date: date | None = None
    responsible_id: int | None = None


class IssueResponse(BaseModel):
    id: int
    folio: str
    title: str
    description: str
    type: str
    priority: str
    status: str
    resolution: str | None
    report_date: date
    commitment_date: date | None
    project_id: int
    responsible_id: int | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/issue.ts
/**
 * Issue/Action/Decision type definitions.
 * Aligned with backend schemas: IssueResponse.
 */

/** Matches backend IssueResponse schema */
export interface Issue {
  id: number;
  folio: string;
  title: string;
  description: string | null;
  type: string;
  priority: string;
  status: string;
  resolution: string | null;
  report_date: string | null;
  commitment_date: string | null;
  project_id: number;
  responsible_id: number | null;
  created_at: string;
}

/** Cross-project issue view used in IssuesPage (with project info) */
export interface IssueWithProject {
  id: number;
  folio: string;
  title: string;
  type: string;
  priority: string;
  status: string;
  project_name: string;
  project_id: number;
  report_date: string;
  commitment_date: string;
}
